arc/conformance/plugin/Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM k8s.gcr.io/build-image/debian-base:bullseye-v1.1.0
+ARG KUBE_VERSION=v1.21.2
+ARG TARGETARCH
+
+# install dependencies
+RUN apt-get update -y && apt-get upgrade -y
+RUN DEBIAN_FRONTEND=noninteractive TZ=America/Los_Angeles apt-get install python3-pip bash -y
+RUN python3 -m pip install junit_xml
+
+# install azcli
+RUN apt-get update && apt-get -y --allow-change-held-packages upgrade && \
+    apt-get -f -y --allow-change-held-packages install curl apt-transport-https lsb-release gnupg python3-pip && \
+    curl -sL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > /etc/apt/trusted.gpg.d/microsoft.asc.gpg && \
+    CLI_REPO=$(lsb_release -cs) && \
+    echo "deb [arch=${TARGETARCH}] https://packages.microsoft.com/repos/azure-cli/ ${CLI_REPO} main" \
+    > /etc/apt/sources.list.d/azure-cli.list && \
+    apt-get update && \
+    apt-get install -y azure-cli && \
+    rm -rf /var/lib/apt/lists/*
+
+# install kubectl
+RUN curl -LO https://storage.googleapis.com/kubernetes-release/release/${KUBE_VERSION}/bin/linux/${TARGETARCH}/kubectl && \
+    chmod +x kubectl && \
+    mv kubectl /usr/local/bin/kubectl
+
+# install step cli to create self signed certificates
+COPY --from=smallstep/step-cli:0.18.0 /usr/local/bin/step /usr/local/bin/step
```

Comment:
can you add an arg for step version too?

Reply:
![image](https://user-images.githubusercontent.com/1626598/151789212-79e2f071-27bb-4f78-8027-cd9e6c867a9b.png)

It should pull right version of ARC

Reply:
The ARG does not evaluate in `COPY --from`. Tried few diff methods like setting up as global arg, exporting it to env var to carry on to from statement etc. but no luck.

Reply:
Dug a little more into this and this dockerfile works:

```bash
ARG STEP_VERSION=0.18.0
ARG STEP_IMAGE=smallstep/step-cli:${STEP_VERSION}
FROM $STEP_IMAGE as step

FROM k8s.gcr.io/build-image/debian-base:bullseye-v1.1.0
ARG KUBE_VERSION=v1.21.2
ARG TARGETARCH

# install dependencies
RUN apt-get update -y && apt-get upgrade -y
RUN DEBIAN_FRONTEND=noninteractive TZ=America/Los_Angeles apt-get install python3-pip bash curl apt-transport-https lsb-release gnupg -y --allow-change-held-packages
RUN python3 -m pip install junit_xml

# install azcli
RUN curl -sL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > /etc/apt/trusted.gpg.d/microsoft.asc.gpg && \
    CLI_REPO=$(lsb_release -cs) && \
    echo "deb [arch=${TARGETARCH}] https://packages.microsoft.com/repos/azure-cli/ ${CLI_REPO} main" \
    > /etc/apt/sources.list.d/azure-cli.list && \
    apt-get update && \
    apt-get install -y azure-cli && \
    rm -rf /var/lib/apt/lists/*

# install kubectl
RUN curl -LO https://storage.googleapis.com/kubernetes-release/release/${KUBE_VERSION}/bin/linux/${TARGETARCH}/kubectl && \
    chmod +x kubectl && \
    mv kubectl /usr/local/bin/kubectl

# install step cli to create self signed certificates
COPY --from=step /usr/local/bin/step /usr/local/bin/step

COPY arc/conformance/plugin/arc_conformance.sh /arc/arc_conformance.sh
COPY arc/conformance/plugin/setup_failure_handler.py /arc/setup_failure_handler.py
COPY test/e2e/_output/${TARGETARCH}/e2e /arc/e2e

RUN ["chmod", "+x", "/arc/arc_conformance.sh"]
ENTRYPOINT ["/arc/arc_conformance.sh"]
```

Having the version, image name and `FROM` in global

Reply:
Updated